client/dfclient/include/text_creator.hpp

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include <ncine/Sprite.h>
+#include "resources.hpp"
+#include "gameplay_state.hpp"
+
+/** Simplifies creating various in-game texts, such as UI messages on player kills.
+ * Designed to allow long-term and one-off text parameter customization. As such,
+ * stores the default values, which are customizable directly or through setter methods
+ * and are supplied if no value for a parameter is given.
+ * 
+ * To simplify creating outlines, this manager also stores the values used to create
+ * the last text. The values are supplied automatically to the outline-creating method,
+ * with the exception of tween easing, which has to be given once more as a parameter.
+ */
+class TextCreator {
+	std::reference_wrapper<Resources> _resources; // to be able to use fonts and emplace tweens
+public:
+	TextCreator(Resources& resources);
+
+	// public for convenience
+	ncine::Color _textColor;
+	float _pos_x;
+	float _pos_y;
+	float _textScale;
+	int _from, _hold, _decay;
+	Layers _layer;
+
+	std::string _lastText;
+	ncine::Color _lastColor;
+	float _last_x;
+	float _last_y;
+	float _lastScale;
+	int _last_from, _last_hold, _last_decay;
+
+	// existent for code clarity
+	inline void setColor(ncine::Color color) { _textColor = color; }
+	inline void setColor(int r, int g, int b, int a = 255) { _textColor = ncine::Color(r,g,b,a); }
+
+	inline void setPosition(float x, float y) { _pos_x = x; _pos_y = y; }
+	inline void setScale(float scale) { _textScale = scale; }
+	inline void setFrom(int from) { _from = from; }
+	inline void setHold(int hold) { _hold = hold; }
+	inline void setDecay(int decay) { _decay = decay; }
+	inline void setTweenParams(int from, int hold, int decay) { setFrom(from); setHold(hold); setDecay(decay); }
```

Comment:
The name of this function is really nondescript. IMO it's better to call those methods separately. Or, rename it to `setFromHoldDelay`?

Reply:
For me, this name is more descriptive than `setFromHoldDelay`, because it tells what the values are actually used for :P It requires knowledge of the TextCreator's workings, but not more than it is required to use it, I feel. If one prefers to use the separate methods, well - that's why they are provided :P